projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/linked-cases/write-linked-cases-field.component.ts

```diff
@@ -89,8 +88,8 @@ export class WriteLinkedCasesFieldComponent extends AbstractFieldWriteComponent
     }
   }
 
-  public getLinkedCaseReasons(): void {
-    const reasonCodeAPIurl = `${this.appConfig.getRDCommonDataApiUrl()}/lov/categories/CaseLinkingReasonCode`;
+  public getLinkedCaseReasons(serviceId: number): void {
+    const reasonCodeAPIurl = `${this.appConfig.getRDCommonDataApiUrl()}/lov/categories/CaseLinkingReasonCode?serviceId=${serviceId}`;
```

Comment:
I think you'll have to change the signature to accept an array and call lov/categories for each service ID - though it's worth checking if the parameter accepts a comma separated list (check the source of the lov api in github

Reply:
@andywilkinshmcts I require more clarification from you on this. Better to chat over the phone